src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java

```diff
@@ -128,12 +129,19 @@ public void run() {
             if (build instanceof AbstractBuild) {
                 AbstractBuild abstractBuild = (AbstractBuild)build;
                 Node node = abstractBuild.getBuiltOn();
-                nodeName = node.getNodeName();
+                if (node != null) {
+                    nodeName = node.getNodeName();
+                }
             }
             int timeZoneOffset = TimeZone.getDefault().getRawOffset();
             String master;
 
-            String result = build.getResult().toString();
+
+            String result = "Unknown";
+            final Result buildResult = build.getResult();
+            if (buildResult != null) {
```

Comment:
Maybe result should be "Running" according to the getResult() docs

Reply:
This code shouldn't execute until after the run is done (for now) so a `null` here would mean unknown and something is wrong, but I don't want to skip logging the statistics just for that.

Reply:
Any Build Step may actually change this expectation. Better safe than sorry